### DISCUSSION TITLE
What should we name the `europa` <-> `cloak` compat layer?

### DISCUSSION THREAD

**Discussion by jlongtine** [dagger/dagger#3161] [General]
And where should it live?

Should it be a part of the new `dagger` binary? Or should it live in a separate binary and repo?

**Comment by shykes** [comment]
* Name: once cloak is GA, the official name of the europa compat layer will be "Dagger Classic". This name should be used in docs, marketing material etc.
* If we do ship Dagger Classic as a separate binary, the natural name would be `dagger-classic`. We can discuss variations but this should be the baseline (because it's the least surprising).
* We may (TBD) ship cloak preview as an experimental feature flag in the current dagger binary. In that case, it may make sense to continue down the same path even after cloak goes GA, and reverse roles: europa becomes a feature flag. The result would effectively be Dagger Classic being bundled into the `dagger` binary.

cc @samalba who suggested the feature flag approach, and @aluzzardi who architected it for past releases.

**Reply by jlongtine** [reply]
@shykes What about the possibility of a `cue` SDK, so that users can use other `cloak` extensions? Would that get wrapped into `dagger-classic`? 

Part of my interest in naming it `dagger-cue` has to do with keeping the door open for that to happen down the road.

**Reply by shykes** [reply]
I think a CUE sdk would be a separate feature with a different scope. If and when we ship a CUE sdk we can reuse as much code as we want from classic. But Classic does not aim to be a complete cue sdk, and it's important IMO to make that clear so that there is no confusion or disappointment.

**Reply by jlongtine** [reply]
Sounds good, @shykes.

**Comment by marcosnils** [comment]
:wave: my two cents from our discord initial thoughts:

is the main argument to have a separate binary so we can iterate on both projects with different cadence? If that's the case kind of SGTM. 

maybe an approach to simplify the user's life is that the dagger binary detects if the cue support is needed and handles the binary download / update for the platform of the user automatically?
we could even ship the dagger-cue thing as a Go plugin (https://pkg.go.dev/plugin) so we don't have to deal with the complexity of building and managing multiple binaries. cc @aluzzardi 

> * We may (TBD) ship cloak preview as an experimental feature flag in the current dagger binary. In that case, it may make sense to continue down the same path even after cloak goes GA, and reverse roles: europa becomes a feature flag. The result would effectively be Dagger Classic being bundled into the `dagger` binary.

After reading this I agree that having a single binary is the simplest approach to start with. If release cadence becomes a problem, we can switch to a different binary / plugin approach later.

**Comment by mircubed** [comment]
@jlongtine since it seems like we have an answer then should we transfer this discussion into an issue?

**Reply by jlongtine** [reply]
I'd like to get a 👍🏻 from @aluzzardi before we call it.
